src/components/input-switch/foreignkey-dropdown-field.tsx

```diff
@@ -0,0 +1,472 @@
+    logObj.action = LogService.getActionString(LogActions.LOAD, stackPath);
+
+    ref.read(initialPageLimit, logObj).then((page: any) => {
+      setCurrentDropdownPage(page);
+      setDropdownRows(page.tuples);
+
+      // if we use props.foreignKeyData.current[props.name], we get an object of row values (tuple.data)
+      // we don't know which column value is used for the displayname so it's better to check react-hook-form state
+      const displayedValue = getValues()[props.name];
+
+      // check if we have a value set for the foreign key input
+      if (displayedValue) {
+        // set the checked row if it's present in the page of rows
+        page.tuples.forEach((tuple: any) => {
+          if (tuple.displayname.value === displayedValue) setCheckedRow(tuple);
+        });
+      }
+
+      setShowSpinner(false);
+      setDropdownInitialized(true);
+    }).catch((exception: any) => {
+      setShowSpinner(false);
+      dispatchError({ error: exception });
+    })
+  }
+
+  /**
+   * make sure the underlying raw columns as well as foreignkey data are also emptied.
+   */
+  const onClear = () => {
+    // clear the raw values
+    props.columnModel.column.foreignKey.colset.columns.forEach((col: any) => {
+      setValue(`${usedFormNumber}-${col.name}`, '');
+    });
+
+    // clear the foreignkey data
+    if (props.foreignKeyData && props.foreignKeyData.current) {
+      props.foreignKeyData.current[props.name] = {};
+    }
+
+    // the input-field will take care of clearing the displayed rowname.
+  }
+
+  // function for fetching data in dropdown after a search term
+  const searchCallback = (value: string | null, action: LogActions) => {
+    const searchRef = dropdownReference.search(value);
+
+    // create initial stack
+    const currStackNode = LogService.getStackNode(LogStackTypes.FOREIGN_KEY, searchRef.table);
+    const stack = LogService.addExtraInfoToStack(LogService.getStackObject(currStackNode), { dropdown: 1 })
+
+    const searchStack = value ? LogService.addExtraInfoToStack(LogService.getStackObject(currStackNode), { dropdown: 1, search_str: value }) : stack;
+
+    // use the action from returned from SearchInput component
+    LogService.logClientAction({
+      action: LogService.getActionString(action, stackPath),
+      stack: searchStack
+    }, searchRef.defaultLogInfo);
+
+    // different action for read
+    // add causes to stack only for read request
+    const logObj = {
+      action: LogService.getActionString(LogActions.RELOAD, stackPath),
+      stack: LogService.addCausesToStack(stack, [LogReloadCauses.DROPDOWN_SEARCH_BOX], ConfigService.ERMrest.getElapsedTime())
+    }
+
+    searchRef.read(pageLimit, logObj).then((page: any) => {
+      setCurrentDropdownPage(page);
+
+      if (page.length === 0) {
+        const fakeTuple = {
+          rowName: {
+            value: '<i>No Results</i>',
+            isHTML: true
+          },
+          uniqueId: 'no_results_row'
```

Comment:
Instead of this, let's just add a special flag to this tuple, similar to how we did this in faceting. Something like `isNoResultRow: true` or a better name

Reply:
Or simply set the `dropdownRows` to empty array and in `renderDropdownOptions` render the special "no results" row if this array is empty. You already have a special handling for it in that function, there's no reason to do something special here.

Reply:
I think at some point I had a function that was rendering each row and I didn't do anything special for the no results row. That eventually changed to having a conditional that I didn't think about cleaning up.